### PR TITLE
[+] CORE : Module manager tests

### PR DESCRIPTION
**Pull request by Quetzacoalt91**

### CODE REVIEW

**Comment by mickaelandrieu** [comment]
Hi @Quetzacoalt91, be careful of changes introduced by https://github.com/PrestaShop/PrestaShop/pull/5229/files :)

**Comment by Quetzacoalt91** [comment]
This pull-request has been opened just to check the travis result and won't be merged today.

**Comment by Quetzacoalt91** [comment]
homer-simpson_woohoo

Tests have passed !

**Comment by mickaelandrieu** [comment]
:sunglasses:

File: src/Adapter/Module/AdminModuleDataProvider.php [hunk 1]
```diff
@@ -228,13 +218,13 @@ protected function applyModuleFilters(array $products, $categories, array $filte
             $search_result = [];
 
             switch ($filter_name) {
-                case 'category':
+                /*case 'category':
```
Comment (mickaelandrieu) [inline src/Adapter/Module/AdminModuleDataProvider.php]: can you remove commented code ?

File: src/Core/Addon/AddonListFilter.php [hunk 2]
```diff
@@ -31,23 +31,78 @@ class AddonListFilter
...
+    public function hasStatus($status)
+    {
+        return ($this->status & $status);
+    }
+
+    public function hasType($type)
+    {
+        return ($this->type & $type);
+    }
+
```
Comment (mickaelandrieu) [inline src/Core/Addon/AddonListFilter.php]: remove some extra lines please :)

File: src/Core/Addon/AddonListFilter.php [hunk 3]
```diff
@@ -31,23 +31,78 @@ class AddonListFilter
...
+    {
+        return $this->addStatus(~$status);
+    }
+
+    public function removeType($type)
+    {
+        return $this->addType(~$type);
+    }
+
+
```
Comment (mickaelandrieu) [inline src/Core/Addon/AddonListFilter.php]: same here

File: src/Core/Addon/Module/ModuleRepository.php [hunk 4]
```diff
@@ -339,6 +352,15 @@ public function getModule($name)
     }
 
     /**
+     * Function which returns the modules directory. Used for mock in tests/ folder.
+     * @return string The modules directory
+     */
+    protected function getModulesDir()
```
Comment (mickaelandrieu) [inline src/Core/Addon/Module/ModuleRepository.php]: wow please :'(

you can "mock" constants using phpunit configuration (https://phpunit.de/manual/current/en/phpunit-book.html#appendixes.configuration.php-ini-constants-variables) or in beforeClass/setUp() functions.

File: tests/Unit/Adapter/Module/AdminModuleDataProviderTest.php [hunk 5]
```diff
@@ -111,7 +111,7 @@ public function test_modules_in_categories_can_be_found_in_module_list()
         }
     }
 
-    public function test_no_results()
+    /*public function test_no_results()
```
Comment (mickaelandrieu) [inline tests/Unit/Adapter/Module/AdminModuleDataProviderTest.php]: can you uncomment tests ?

File: tests/Unit/Core/Addon/Module/ModuleRepositoryTest.php [hunk 6]
```diff
@@ -0,0 +1,299 @@
...
+use PrestaShop\PrestaShop\Core\Addon\AddonListFilter;
+use PrestaShop\PrestaShop\Core\Addon\AddonListFilterOrigin;
+use PrestaShop\PrestaShop\Core\Addon\AddonListFilterStatus;
+use PrestaShop\PrestaShop\Core\Addon\AddonListFilterType;
+use PrestaShop\PrestaShop\Core\Addon\Module\ModuleManagerBuilder;
+use PrestaShop\PrestaShop\Tests\Fake\FakeLogger;
+use PrestaShop\PrestaShop\Tests\TestCase\UnitTestCase;
+
+/**
+ * @runTestsInSeparateProcesses
```
Comment (mickaelandrieu) [inline tests/Unit/Core/Addon/Module/ModuleRepositoryTest.php]: what do you need this ?

File: tests/Unit/Core/Addon/Module/ModuleRepositoryTest.php [hunk 7]
```diff
@@ -0,0 +1,299 @@
...
+ * @preserveGlobalState disabled
+ */
+class ModuleRepositoryTest extends UnitTestCase
+{
+
```
Comment (mickaelandrieu) [inline tests/Unit/Core/Addon/Module/ModuleRepositoryTest.php]: remove extra line

File: tests/resources/modules/ps_featuredproducts/Readme.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
+# Featured products
```
Comment (mickaelandrieu) [inline tests/resources/modules/ps_featuredproducts/Readme.md]: we may dont need this file ^^

**Comment by mickaelandrieu** [comment]
thank you @Quetzacoalt91

**Pull request merged**